dogs/src/main/java/com/airbnb/mvrx/dogs/DogViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.airbnb.mvrx.dogs
+
+import com.airbnb.mvrx.MvRxViewModelFactory
+import com.airbnb.mvrx.ViewModelContext
+import com.airbnb.mvrx.dogs.app.DogApplication
+import com.airbnb.mvrx.dogs.app.MvRxViewModel
+import com.airbnb.mvrx.dogs.data.DogRepository
+import io.reactivex.schedulers.Schedulers
+
+class DogViewModel(
+    state: DogState,
+    private val dogRepository: DogRepository
+) : MvRxViewModel<DogState>(state) {
+
+    init {
+        dogRepository.getDogs()
+            .subscribeOn(Schedulers.io())
```

Comment:
Nit: move `subscribeOn(Schedulers.io())` to the end of `getDogs` function inside repository